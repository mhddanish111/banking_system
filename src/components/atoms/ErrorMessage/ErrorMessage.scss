@import '../../../styles/colors.scss';
@import '../../../styles/spacing.scss';
.error-message-container {
    text-align: center;
    // margin: 0 ($space-l - $space-xxs);
    margin-bottom: $space-xs;
    padding: $space-xxs;
    color: white;
  .close-btn {
    margin-left: 15px;
    color: white;
    font-weight: bold;
    float: right;
    font-size: 22px;
    line-height: 20px;
    cursor: pointer;
    transition: 0.3s;
  }

  .close-btn:hover {
    color: black;
  }
  &.alert {
    background-color: $color-error;
  }
  &.success {
    background-color: $color-success;
  }
}
