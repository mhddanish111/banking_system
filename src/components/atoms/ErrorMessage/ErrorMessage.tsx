import React, { ReactNode } from 'react';
import './ErrorMessage.scss';

type ErrorProps = {
  className?: string;
  message?: string;
  children?: ReactNode;
  onClick?: (e: any) => void;
};
const ErrorMessage: React.FC<ErrorProps | any> = (props) => {
  const { className, message, onClick, children } = props;
  return (
    message && (
      <div className={`error-message-container ${className}`}>
        <div className="message-container">
          <span className="close-btn" onClick={(e) => onClick(e)}>
            &times;
          </span>
          {message}
        </div>
        <div className="child-container">{children}</div>
      </div>
    )
  );
};

export default ErrorMessage;
