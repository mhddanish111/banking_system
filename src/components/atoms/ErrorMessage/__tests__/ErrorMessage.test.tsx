import React from 'react';
import { shallow } from 'enzyme';
import ErrorMessage from '../ErrorMessage';

describe('<Error />', () => {
  const props = {
    className: '',
    message: 'Something went wrong',
    children: <div />,
    onClick: jest.fn(),
  };
  const wrapped = shallow(<ErrorMessage {...props} />);
  it('should render the Error Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should call onClick event', () => {
    const onClick = jest.fn();
    const event = {
      preventDefault: jest.fn(),
    };
    wrapped.find('.close-btn').simulate('click', onClick(event));
    expect(onClick).toHaveBeenCalled();
  });
});
