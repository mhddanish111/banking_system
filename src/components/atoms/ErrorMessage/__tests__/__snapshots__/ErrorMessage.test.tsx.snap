// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render the Error Component correctly 1`] = `
<div
  className="error-message-container "
>
  <div
    className="message-container"
  >
    <span
      className="close-btn"
      onClick={[Function]}
    >
      Ã—
    </span>
    Something went wrong
  </div>
  <div
    className="child-container"
  >
    <div />
  </div>
</div>
`;
